test/controllers/api/v2/hosts_controller_test.rb

```diff
@@ -1230,4 +1230,36 @@ def setup
     assert_response :success
     assert_equal puppet_proxy['name'], JSON.parse(@response.body)['puppet_proxy']['name'], "Can't update host with puppet proxy #{puppet_proxy}"
   end
+
+  # This is a test of the base_controller functionality, but we need to use a real endpoint
+  # to test the API response metadata is returned correctly
+  describe 'should create correct subtotals' do
+    before do
+      @empty_org = Organization.find_by_name("Empty Organization")
+      as_admin do
+        FactoryBot.create_list(:host, 100, organization: @empty_org)
```

Comment:
instead of creating 100 entries, you can create a smaller list (say 10) and pass a smaller per_page parameter in the request (say 5)